### PR TITLE
IDN: Use TR46 'transitional' for toASCII translations

### PR DESCRIPTION
**Pull request by rockdaboot**
References: http://unicode.org/faq/idn.html
            http://unicode.org/reports/tr46

### CODE REVIEW

File: lib/url.c [hunk 1]
```diff
@@ -4010,7 +4010,12 @@ static void fix_hostname(struct connectdata *conn, struct hostname *host)
 #ifdef USE_LIBIDN2
     if(idn2_check_version(IDN2_VERSION)) {
       char *ace_hostname = NULL;
-      int rc = idn2_lookup_ul((const char *)host->name, &ace_hostname, 0);
+#ifdef IDN2_TRANSITIONAL
+      int flags = IDN2_INPUT_NFC | IDN2_TRANSITIONAL;
```
Comment (bagder) [inline lib/url.c]: I was a bit too quick. Which version adds `IDN2_INPUT_NFC`? It seems we need to check for that as well, right?

Comment (rockdaboot) [inline lib/url.c]: No. AFAICS, NFC support was there from the beginning (used git grep).

Comment (bagder) [inline lib/url.c]: You mean `IDN2_NFC_INPUT` then, right? =)

Comment (rockdaboot) [inline lib/url.c]: Just amend the commit.... sorry, I pushed the wrong one (now corrected).
`IDN2_INPUT_NFC` must be `IDN2_NFC_INPUT`

**Comment by bagder** [comment]
Thanks a lot!

**Comment by rockdaboot** [comment]
OMG... I just started to add similar code to wget... IDN2_* are enums and not defines (can't say *how* much I hate enums in public headers). So we have to use the version number, like
```#if IDN2_VERSION_NUMBER >= 0x00140000```

I'll prepare a second pull request soon...

**Pull request closed without merging**